### DISCUSSION TITLE
Motivation for writing Rio?

### DISCUSSION THREAD

**Discussion by freeo** [raphamorim/rio#76] [General]
@raphamorim Why did you write Rio? 

I couldn't find any info about your motivation to do so neither on your website, the repo nor the issues.

I'm always curious about new terminals and I guess everyone stumbling upon it would also be interested in your goals/roadmap/motivation with this project.

**Comment by raphamorim** [comment]
Hey there @freeo ,

Thanks for bringing this up. Rio originally started as my personal project to study few things and ended up evolving to be my main terminal. The reason Rio became an OSS project was due to keep track of my studies and the fact of the more I used GPU based terminals more I felt there's a lot of possibilities to explore that haven't been explored yet. For example explore usage of virtual terminals in the Web, allow usage of WASM to write widgets by any language (e.g making your own implementation of tabs), custom protocols for render more than images (#38) and other things that I wanted (and still want) to explore.

While all these things are attractive and would love to work on it right way, I understand that is a requirement have the terminal in a stable state (usable by multi platforms) to start move ideas to the practical side, that's why haven't added future goals in the `README.md` or website. Rio is still a new-born project and I don't want limit possible ideas coming from external contributors by writing goals in a stone for an unstable project. Once Rio pass the stabilization phase I will announce a proper (more clear) roadmap.
